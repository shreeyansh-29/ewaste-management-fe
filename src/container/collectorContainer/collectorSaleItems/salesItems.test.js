/* eslint-disable no-undef */
import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, {mount, shallow} from "enzyme";
import SalesItems from "./salesItems";
import toJson from "enzyme-to-json";
import {Tabs} from "@mui/material";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

Enzyme.configure({adapter: new Adapter()});

describe("test salesItems", () => {
  it("should render SalesItems", () => {
    const wrapper = shallow(<SalesItems />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should test onChange", () => {
    const onChange = jest.fn();
    const event = {
      preventDefault() {},
    };
    const wrapper = shallow(<SalesItems handleChange={onChange} />);
    wrapper.find(Tabs).simulate("change", event);
    expect(onChange).toHaveBeenCalled;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  let store;
  store = mockStore({
    customerPendingRequest: {
      data: {
        status: "success",
        data: {
          id: 1,
        },
      },
    },
  });
  it("should test TabPanel", () => {
    const name = "TabPanel";
    const props = {children: "", index: "", other: ""};
    const TabPanel = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SalesItems {...props} TabPanel={TabPanel}>
            {name}
          </SalesItems>
        </BrowserRouter>
      </Provider>
    );
    wrapper.find(TabPanel).exists();
    expect(TabPanel).toHaveBeenCalled;
  });
});
