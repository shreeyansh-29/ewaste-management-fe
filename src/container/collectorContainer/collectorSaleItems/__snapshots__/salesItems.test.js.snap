// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test salesItems should render SalesItems 1`] = `
<div
  style={
    Object {
      "padding": "150px 30px 0 30px",
    }
  }
>
  <h2
    style={
      Object {
        "backgroundColor": " rgb(30, 28, 54)",
        "borderRadius": "5px",
        "color": "white",
        "fontSize": "30px",
        "marginBottom": "2.5%",
        "padding": "2px,",
        "textAlign": "center",
      }
    }
  >
     
    Sales Summary
     
  </h2>
  <ForwardRef(Paper)
    square={true}
    style={
      Object {
        "margin": "20px auto",
        "width": "100%",
      }
    }
  >
    <ForwardRef(Tabs)
      aria-label="disabled tabs example"
      centered={true}
      elevation={20}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
      variant="fullWidth"
    >
      <ForwardRef(Tab)
        label="Available"
      />
      <ForwardRef(Tab)
        label="Sold"
      />
    </ForwardRef(Tabs)>
    <TabPanel
      index={0}
      value={0}
    >
      <Connect(AvailableSales) />
    </TabPanel>
    <TabPanel
      index={1}
      value={0}
    >
      <Connect(SoldItems) />
    </TabPanel>
  </ForwardRef(Paper)>
</div>
`;

exports[`test salesItems should test onChange 1`] = `
<div
  style={
    Object {
      "padding": "150px 30px 0 30px",
    }
  }
>
  <h2
    style={
      Object {
        "backgroundColor": " rgb(30, 28, 54)",
        "borderRadius": "5px",
        "color": "white",
        "fontSize": "30px",
        "marginBottom": "2.5%",
        "padding": "2px,",
        "textAlign": "center",
      }
    }
  >
     
    Sales Summary
     
  </h2>
  <ForwardRef(Paper)
    square={true}
    style={
      Object {
        "margin": "20px auto",
        "width": "100%",
      }
    }
  >
    <ForwardRef(Tabs)
      aria-label="disabled tabs example"
      centered={true}
      elevation={20}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      variant="fullWidth"
    >
      <ForwardRef(Tab)
        label="Available"
      />
      <ForwardRef(Tab)
        label="Sold"
      />
    </ForwardRef(Tabs)>
    <TabPanel
      index={0}
    >
      <Connect(AvailableSales) />
    </TabPanel>
    <TabPanel
      index={1}
    >
      <Connect(SoldItems) />
    </TabPanel>
  </ForwardRef(Paper)>
</div>
`;
